app/models/aleph_item.rb

```diff
@@ -72,3 +83,407 @@ def xml_status(id)
+    return unless @oclc_number.present?
+    "https://mit.worldcat.org/search?q=no%3A#{@oclc_number}"
+  end
+
+  # special_ill items are only eligible via BorrowDirect.
+  def url_for_special_ill
+    # @identifier may be an ISBN or the ISSN, but either one works in the BD
+    # ISBN parameter.
+    "https://library.mit.edu/shib/bd.cgi?url_ver=Z39.88-2004&amp;rft.isbn=#{@identifier}"
+  end
+
+  # ~~~~~~~~ Utility functions needed to determine PDF scan eligibility ~~~~~~~~
+  def call_number_valid_for_scan?
+    !@call_number.start_with?('ARCH DRAW', 'ATLAS', 'AUDIO', 'AUDTAPE', 'CD',
+                              'CDROM', 'DSKETTE', 'DVD', 'DVDROM', 'FICHE',
+                              'FILM', 'FOLIO', 'INDEX', 'MAP', 'MFILM',
+                              'OVRSIZE', 'RECORD', 'REGULAR', 'SCORE', 'SMALL',
+                              'THESIS', 'USB DRIVE', 'VDISC', 'VIDEO')
+  end
+
+  def z30status_valid_for_scan?
+    [
+      '60 Day Loan',
+      'One Week Loan',
+      '1 Week No Renew',
+      'OCC 60',
+      'LSA 7',
+      'LSA 60',
+      'LSA Use Only'
+    ].include? @z30status
+  end
+
+  def collection_valid_for_scan?
+    [
+      'Stacks',
+      'Journal Collection',
+      'Off Campus Collection',
+      'Science Journals',
+      'Humanities Journals',
+      'Impulse Borrowing Display',
+      'Browsery',
+      'Graphic Novel Collection',
+      'Travel Collection'
+    ].include? @collection
+  end
+
+  def status_valid_for_scan?
+    ['In Library', 'New Books Displ'].include?(@status)
+  end
+
+  def library_valid_for_scan?
+    ['Physics Dept. Reading Room', 'Rotch Visual Collections',
+     'Institute Archives'].exclude? @library
+  end
+
+  # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Misc utilities ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  # Items with these process & item statuses may be put on hold/recalled.
+  def requestable?
+    return false if @on_reserve
+    [
+      # Items with these status codes may be requested from any library,
+      # except the Annex.
+      ['01', '03', '04', '05', '06', '07', '08', '09', '11', '12', '19', '20',
+       '21'].include?(@z30status_code) && @library != 'Library Storage Annex',
+
+      # Items with status code 23 may be requested from only some libraries.
+      @z30status_code == '23' && ['Barker Library',
+                                  'Dewey Library',
+                                  'Hayden Library',
+                                  'Rotch Library',
+                                  'Library Storage Annex'].include?(@library),
+
+      # The Annex is special.
+      @library == 'Library Storage Annex' && ['13', '14', '15', '56',
+                                              '57'].include?(@z30status_code)
+    ].any?
+  end
+
+  def host_values
+    if Rails.env.production?
+      sfx_host = 'https://sfx.mit.edu/sfx_local'
+      request_host = 'library.mit.edu'
+    else
+      sfx_host = 'https://sfx.mit.edu/sfx_test'
+      request_host = 'walter.mit.edu'
```

Comment:
We should use prod SFX for all all environments.

Reply:
Done.